File: Leetcode/165 - Compare Version Numbers/solution.ts
function compareVersion(version1: string, version2: string): number {
    const v1: number[] = version1.split('.').map(Number);
    const v2: number[] = version2.split('.').map(Number);

    const len: number = Math.max(v1.length, v2.length);
    for (let i = 0; i < len; i++) {
        const num1: number = v1[i] ?? 0;
        const num2: number = v2[i] ?? 0;
        if (num1 !== num2) return num1 > num2 ? 1 : -1;
    }

    return 0;
};

File: Leetcode/165 - Compare Version Numbers/solution.py
def compareVersion(version1: str, version2: str) -> int:
    v1: List[int] = [int(x) for x in version1.split('.')]
    v2: List[int] = [int(x) for x in version2.split('.')]

    length: int = max(len(v1), len(v2))
    for i in range(length):
        num1: int = v1[i] if i < len(v1) else 0
        num2: int = v2[i] if i < len(v2) else 0
        if num1 != num2:
            return 1 if num1 > num2 else -1
    return 0